engine/src/main/java/org/hibernate/validator/internal/engine/messageinterpolation/ElTermResolver.java

```diff
@@ -79,7 +85,26 @@ public String interpolate(MessageInterpolator.Context context, String expression
 		return resolvedExpression;
 	}
 
-	private ValueExpression bindContextValues(String messageTemplate, MessageInterpolator.Context messageInterpolatorContext, SimpleELContext elContext) {
+	private ELContext getElContext(MessageInterpolator.Context context) {
+		if ( !( context instanceof HibernateMessageInterpolatorContext ) ) {
+			return new SafestELContext( expressionFactory );
+		}
+
+		switch ( ( (HibernateMessageInterpolatorContext) context ).getExpressionLanguageFeatureLevel() ) {
+			case NONE:
+				throw LOG.expressionsNotResolvedWhenExpressionLanguageFeaturesDisabled();
+			case SAFEST:
+				return new SafestELContext( expressionFactory );
```

Comment:
If you want to be consistent will `INSECURE`, this should be named `SECURE`. Safety and security not being the same thing and all that...

Also, technically, the most secure ELContext is no ELContext... If there is a way to leak data from resource bundles, or to access a property of a variable, we cannot really say "it's absolutely secure".

To avoid the problem, isn't there a name that translates what the context allows, rather than our opinion of it? `VARIABLES`, maybe?